chartfx-generate/src/main/java/de/gsi/generate/CodeGenerator.java

```diff
@@ -0,0 +1,236 @@
+package de.gsi.generate;
+
+import static java.nio.file.StandardOpenOption.*;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.*;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * Proof of concept code generator that gets called by the maven exec plugin during the generate
+ * sources phase. It checks the source files for code gen templates by assuming some naming convention,
+ * and then does search/replace based on sections within the source file. This allows multiple
+ * output types to be generated into a single class.
+ *
+ * Currently the generated class inherits from the base class, but I'm not sure that's necessary.
+ * We could just generate a full copy of the base class and then add sections within the template
+ * area. TBD.
+ *
+ * @author ennerf
+ * @author Alexander Krimm
+ */
+@Mojo(name = "generate-sources", defaultPhase = LifecyclePhase.GENERATE_SOURCES)
+public class CodeGenerator extends AbstractMojo {
+    @Parameter(defaultValue = "${project}", required = true, readonly = true)
+    MavenProject project;
+
+    @Parameter(defaultValue = "${project.build.sourceDirectory}", required = false, readonly = true)
+    String input;
+    @Parameter(defaultValue = "${project.build.directory}/generated-sources/codegen", required = false, readonly = true)
+    String output;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
```

Comment:
Codacy found an issue: [The method 'execute()' has an NPath complexity of 54184, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)